src/oData/jsonToOData.ts

```diff
@@ -97,7 +132,15 @@ const handlePrimitiveFilter = (
 	return maybePluralOp('$and', filters);
 };
 
-const wrapValue = (parentKeys: string[], value: any): PineFilterObject => {
+const wrapValue = (
+	parentKeys: string[],
+	value: any,
+): PineFilterObject | undefined => {
+	// TODO: Remove me as soon as we have a fix for OData
+	if (typeof value === 'number' && value % 1 !== 0) {
```

Comment:
Wouldn't it be more appropriate if we moved this to v, rather that in the odata converter?
https://github.com/balena-io-modules/autoui/blob/78e4d7b26286f63f619a5bdad86706f0a4e60db0/src/DataTypes/number.tsx#L21-L30

Reply:
since this is a DB bug I was adding the logic to avoid the issue only when server side pagination is applied. Client siede this is not an issue. Having said that let me open the PR to avoid filtering in both cases.